### PR TITLE
Adds Symbol levels description

### LINKED ISSUE #908: Symbols levels is not documented
Symbols levels options is not documented in the Users Manual.

My Idea would be to add a generic subsection in working in vector menu data chapter  in the Style Menu. (below the last renderer, and before layer rendering)

### PR DESCRIPTION
**Pull request by SrNetoChan**
Fixes #908

### CODE REVIEW

**Comment by SrNetoChan** [comment]
Reviews are most appreciated! @DelazJ @yjacolin thanks

**Comment by SrNetoChan** [comment]
Any of you knows what's with the

    .. index:: Inverted_Polygons_Renderer

instead of simple references like:

    .. _inverted_polygons_renderer:

?

**Comment by yjacolin** [comment]
@SrNetoChan this is two different things. The second is an anchor to make a reference. The first aims to create an index page (you know like a book to find where a specific word is used in the text). Not sure if I am clear but they are not currently enabled in the html page. See this direct link: http://docs.qgis.org/testing/en/genindex.html

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 1]
```diff
@@ -600,6 +600,55 @@ units).
...
 
+.. _Symbols_levels:
+
+Symbols levels
+..............
+
+For renderers that allow stacked symbol layers (for example, 
+:ref:`single_symbol_renderer`, :ref:`inverted_polygon_renderer` and
+:ref:`rule_based_rendering`) there is an option to control the rendering order
+of each symbol level.
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: plural? level**s**

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 2]
```diff
@@ -600,6 +600,55 @@ units).
...
+.. figure:: /static/user_manual/working_with_vector/symbol_levels.png
+   :align: center
+
+   Symbol levels dialog
+
+.. note::
+
+   If symbols levels are deactivated, the complete symbols will be drawn
+   according to their respective features order. Overlapping symbols will
+   simply ofuscate to other below. Besides, similar symbols won't "merge" with
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: obfuscate

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 3]
```diff
@@ -600,6 +600,55 @@ units).
    The 2.5 D height and angle values are saved in the layer's variables,
    so you can edit it afterwards in the variables tab of the layer's properties dialog.
 
+.. _Symbols_levels:
+
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: could you add: `.. index:: Symbols levels` before `.. _Symbols_levels:`

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 4]
```diff
@@ -600,6 +600,55 @@ units).
...
+Symbols levels
+..............
+
+For renderers that allow stacked symbol layers (for example, 
+:ref:`single_symbol_renderer`, :ref:`inverted_polygon_renderer` and
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: `inverted_polygon_renderer` didn't exist, so I added it directly to master. Could you rebase your PR?

Comment (SrNetoChan) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: Yes, sure.

**Comment by DelazJ** [comment]
LGTM
However I was wondering if it would make sense to have this section near the "Draw Effects" section under the current "Other Setting" section and move them all after the "Feature Rendering" section, still under the "Style menu" section. These two options are some particular tools from the Style menu. In this case, we have:

Vector Properties
--> General menu
--> Style Menu
-----> Features rendering
-------> Single Symbol Renderer
-------> Categorized Renderer
...
...
-------> Heatmap
-------> 2.5 D
-----> Layer rendering
-----> Others Settings
-------> Symbols Levels
-------> Draw Effects
--> Labels menu
What do you think?

**Comment by SrNetoChan** [comment]
Sounds good. I can do that in this PR as well.

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 5]
```diff
@@ -648,17 +648,252 @@ features of the layer:
   used to control the :index:`layer rendering` is retranscribed in the textbox
   beside |checkbox| :guilabel:`Control feature rendering order` option.
 
-.. _figure_symbology_10:
+.. _figure_layer_rendering_1:
 
 .. only:: html
 
-     **Figure Symbology 10:**
+   >**Figure Layer Rendering 1:**
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: typo with >

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 6]
```diff
@@ -648,17 +648,252 @@ features of the layer:
...
+
+.. _Symbols_levels:
+
+Symbols levels
+..............
+
+For renderers that allow stacked symbol layers (for example, 
+:ref:`single_symbol_renderer`, :ref:`inverted_polygon_renderer` and
+:ref:`rule_based_rendering`) there is an option to control the rendering order
+of each symbol's levels.
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: Why not citing categorized nor graduated renderer?
Actually The "Symbols levels" widget has more to do with symbology definition than style renderer imho. Whenever the feature representation is defined using multiple levels of symbols then there's a room to use that option. And all renderers (except heatmap and 2.5D) do allow this (might be weird used with displacement point but sounds possible)

**Comment by SrNetoChan** [comment]
Those were examples. I have choose those because at that time I did not understood the index thing. Now I can choose any other. You are right, and the image is from category renderer. Or should we mention all?

**Comment by SrNetoChan** [comment]
Added references to all renderers sub-section for consistency, fixed the typo, updated how to access each to the symbols levels option for each renderer (rule based and point displacement are different)

**Comment by DelazJ** [comment]
LGTM

**Pull request merged**